src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -244,6 +246,231 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
             }
         }
 
+        internal static ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> GetAnalyzerConfigOptions(
+            IReadOnlyList<string> sourcePaths,
```

Comment:
Why `IReadOnlyList<string>` here? Next method takes an `ImmutableArray<string>` for a paths parameter.

Reply:
This one is used for testing and I pass in an array instead of an ImmutableArray because the syntax is shorter and clearer.